### DISCUSSION TITLE
Difference between `add-data` and `add-binary`

### DISCUSSION THREAD

**Discussion by Blank-c** [pyinstaller/pyinstaller#7370] [PyInstaller] [answered]
What is the difference between `--add-data` and `--add-binary`. You can include a binary using the `--add-data` too, so why is there a different flag for this?

**Comment by rokm** [comment, marked as answer]
"binaries" undergo dependency analysis (i.e., to determine what shared libraries they are using), and additional OS-specific processing (like path rewriting and re-signing on macOS). So sometimes it doesn't matter, and sometimes it does, and even our hooks are not really consistent about it. And there are cases when we intentionally collect a binary as data to avoid the current limitations of the PyInstaller (e.g., shared lib duplication stemming from the dependency analysis).

That said, in the future, I'm planning to implement an identification and re-classification step so the distinction will not matter (but likely both options will remain for backwards compatibility).
